server/middlewares/validators/users.ts

```diff
@@ -317,6 +318,21 @@ const userAutocompleteValidator = [
   param('search').isString().not().isEmpty().withMessage('Should have a search parameter')
 ]
 
+const ensureUserAccountValidator = [
+  async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+    const user = res.locals.oauth.token.User
+    const userAccount = await AccountModel.load(user.Account.id)
```

Comment:
You already have the account in `res.locals.account` so just check `res.locals.account.id === user.Account.id` :)